File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    int divide = 0;
    int tmp_x = x;
    
    bool answer = true;
    
    while(tmp_x > 0) {
        divide += tmp_x%10;
        tmp_x /= 10;
    }
    
    return x%divide == 0 ? true : false;
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    let sum = 0;
    let tmp_x = x;
    
    while (x > 0) {
        sum += x%10;
        x /= 10;
        x = ~~x;
    }
 
    if (tmp_x % sum == 0) return true;
    else return false;
    
}